crates/types/src/nodes_config.rs

```diff
@@ -373,9 +376,9 @@ impl StorageState {
         }
     }
 
-    /// Empty nodes are automatically excluded from node sets.
+    /// Empty nodes are considered not part of the nodeset and they'll never join back.
     pub fn empty(&self) -> bool {
```

Comment:
Can you point me to the place where we classify empty nodes implicitly as authoritative? I could only find `StorageState::is_authoritative()` which bases its decision in the `StorageState`.

Reply:
In EffectiveNodeset. We remove provisioning nodes which implies they are authoritatively empty.

Reply:
This makes a lot of sense now. Thanks for the clarification.